ui/app/components/clients/history.js

```diff
@@ -185,12 +185,19 @@ export default class History extends Component {
     return this.queriedActivityResponse || this.args.model.activity;
   }
 
-  get byMonthTotalClients() {
-    return this.getActivityResponse?.byMonth;
+  get byMonthActivityData() {
+    if (this.selectedNamespace) {
+      return this.filteredActivityByMonth;
+    } else {
+      return this.getActivityResponse?.byMonth;
+    }
   }
 
   get byMonthNewClients() {
-    return this.byMonthTotalClients.map((m) => m.new_clients);
+    if (this.byMonthActivityData) {
+      return this.byMonthActivityData.map((m) => m.new_clients);
```

Comment:
`byMonthActivityData` could return null if `this.getActivityResponse` is undefined since we have optional chaining on line 192, so I'd either return a fallback empty array from `byMonthByActivityData` or do optional chaining here as well

Reply:
good call, thank you!